### DISCUSSION TITLE
Custom pricing, order-only b2b shop, cms functionality

### DISCUSSION THREAD

**Discussion by globalexport** [vendurehq/vendure#2847] [Q&A] [answered]
Hi, 
I am currently researching this ecommerce solution to find out, if it is matching my client's demands. 

Coming from a Shopware instance, we would need a shop which has
1. no pricing without login
2. custom pricing for each logged-in customer and product.
3. no payment process (just ordering and order confirmation emails) 
4. some CMS capabilities

I cannot find email configurations in the demo backend. Is email not implemented? 
There are no custom cms pages visible in the backend. How to give control to the client over writing imprints, terms & conditions, etc?

As it is an headless system, hiding prices from unauthorized visitors seems to be in our control, I assume. But how to set custom pricing for existing b2b customers? Is this possible? 

Is it possible to switch off the payment process (3)? 

Thank you! :)

**Comment by globalexport** [comment]
And where to search the products?

<img width="970" alt="Bildschirmfoto 2024-05-13 um 11 20 33" src="https://github.com/vendure-ecommerce/vendure/assets/70586640/915911d6-3101-4fd2-879a-ef14568bd76f">

**Comment by michaelbromley** [comment, marked as answer]
Hi,

> As it is an headless system, hiding prices from unauthorized visitors seems to be in our control, I assume. But how to set custom pricing for existing b2b customers? Is this possible?

Yes, you can of course hide prices in your front-end logic. But furthermore you can override the way fields like `ProductVariant.price` get resolved so that you can also hide the prices from direct API calls.

To implement customer-specific pricing would require a custom implementation. You'd [create a plugin](https://docs.vendure.io/guides/developer-guide/plugins/) in which you can define the required entities & logic. For instance, you might want to create a new entity to store the customer-specific prices, which has a relation to both a ProductVariant and a Customer. Then you can [extend the ProductVariant type in the GraphQL API](https://docs.vendure.io/guides/developer-guide/extend-graphql-api/#add-fields-to-existing-types) to look up the appropriate price when that customer views the product detail page.

> I cannot find email configurations in the demo backend. Is email not implemented?

Yes, we have the [EmailPlugin](https://docs.vendure.io/reference/core-plugins/email-plugin/) which handles emails.

> Is it possible to switch off the payment process (3)?

Yes, you can fully [customize the order process](https://docs.vendure.io/guides/core-concepts/orders/#custom-order-processes) to suit your needs.

> And where to search the products?

Click the little magnifying glass icon in the top left of the table.

If you have follow-up questions you are very welcome to join our lively [Discord community](https://www.vendure.io/community).

**Comment by globalexport** [comment]
Oh, so many thanks for the eye-opening answers. That really helped a lot.  👍
